code/modules/mining/laborcamp/laborshuttle.dm

```diff
@@ -24,4 +24,5 @@
 		if(S && S.name == "laborcamp_away")
 			to_chat(usr, "<span class='warning'>Shuttle is already at the outpost!</span>")
 			return 0
-	..()
+	..()	//I know the tutorial says if(..()) should always be at the start, but that won't work here
```

Comment:
Don't do this. If you want to add checks before any `tgui_act` in a subclass, create and call a proc called something like `can_call_shuttle` in the parent class (`/obj/machinery/computer/shuttle` I'm guessing). What you're doing right now could break a lot of things and is exploit prone.

Reply:
oh, now that's an approach I hadn't thought of. Thanks!